web_src/js/index.js

```diff
@@ -2167,6 +2167,81 @@ function searchRepositories() {
   });
 }
 
+function hitAllKeys(keys, entry) {
+  let i = 0;
+  let j = 0;
+  const hitIndexes = [];
```

Comment:
I think you can use a [`Set`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set) instead of a Array for better performance.

Reply:
Hi, silverwind

Thanks for your suggestion. 

There isn't too much Insert or Remove operation in the array set.  It just stores the indexes of hit chars in the string. I think we no need to ask for higher performance...

Also, I didn't see anywhere use Set in the web_src/js/index.js

Reply:
My point is that if you're only checking for existance later on  via `.includes`, it's better to just use a Set because Set existance check is O(1) complexity while the same for an array is O(n), so will always be faster unless array length is 1.

Reply:
This algorithm is sub-string matching, it's unlikely to use `Set` here. For example:

```
keys(input) = aaabbbccc
entry(to_match) = abc
```

then the result might be `[a]aa[b]bb[c]cc`